blog/_posts/2025-02-28-afd-aks-ingress-tls-approuting.md

```diff
@@ -0,0 +1,414 @@
+authors: 
+  - steve_griffith
+---
+## Introduction
+
+When running globally distributed public applications in Kubernetes, having access to a global traffic management solution is critical to ensuring high availability and security at the edge. Fortunately, [Azure Front Door](https://learn.microsoft.com/en-us/azure/frontdoor/front-door-overview) provides an easy-to-use global traffic routing capability, with integrated Content Delivery Network and Web Application Firewall. 
+
+Since Azure Front Door is a global resource, it isn't pinned to any one given region or Virtual Network, allowing it to connect to any endpoint globally. With support for [Private Link](https://learn.microsoft.com/en-us/azure/frontdoor/private-link) as an origin target, you can deploy Azure Front Door in-front of private backend services, increasing the overall security of your hosting infrastructure and giving you access to all the capabilities of Front Door.
+
+On the backend, Nginx Ingress is an easy-to-use Kubernetes ingress controller. [AKS Application Routing](https://learn.microsoft.com/en-us/azure/aks/app-routing) provides a managed experience for deploying and managing Nginx Ingress in your AKS clusters. By pairing Azure Front Door with AKS App Routing, you get one of the easiest operational experiences for deploying and managing globally distributed web endpoints with content delivery optimization and security.
+
+In this walkthrough we'll deploy an app with end-to-end TLS encryption, using Azure Front Door as the Internet Facing TLS endpoint and AKS App Routing Ingress as the internal in-cluster ingress controller. 
+
+We'll use Azure Key Vault to store the TLS certificate, and will use App Routing Key Vault integration to get the certificate secret into the ingress controller. 
+
+> *Note:* We'll be walking through step by step, showing the moving parts, via the Azure CLI. If you'd like to see a full implementation of a very similar deployment using Bicep, check out the link below:
+
+[End-to-end TLS with AKS, Azure Front Door, Azure Private Link Service, and NGINX Ingress Controller](https://learn.microsoft.com/en-us/samples/azure-samples/aks-front-door-end-to-end-tls/aks-front-door-end-to-end-tls/)
+
+Let's get to it....
+
+## Network Setup
+
+First, we'll need to establish the network where our AKS cluster will be deployed. Nothing special in our network design. Just a Vnet with a single subnet for the cluster.
+
+```bash
+# Set starting environment variables
+RG=E2ETLSLab
+LOC=eastus2
+
+# Create the Resource Group
+az group create -g $RG -l $LOC
+
+# Get the resource group id
+RG_ID=$(az group show -g $RG -o tsv --query id)
+
+# Set environment variables for the VNet creation
+VNET_NAME=lablab-vnet
+VNET_ADDRESS_SPACE=10.140.0.0/16
+AKS_SUBNET_ADDRESS_SPACE=10.140.0.0/24
+
+# Create the Vnet along with the initial subet for AKS
```

Comment:
spelling on subnet # Create the Vnet along with the initial subet for AKS -> # Create the Vnet along with the initial subnet for AKS

Reply:
Everything else LGTM.